### PR TITLE
RA-1283 Environmental Allergens are not working correctly

### PR DESCRIPTION
**Pull request by reubenvarghese1**
## Related Issue

see https://issues.openmrs.org/browse/RA-1283

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dkayiwa** [comment]
You also need to take care of MigrateAllergiesChangeSet

**Comment by reubenvarghese1** [comment]
@dkayiwa  Thanks for pointing that out. Done :-)

File: api/src/main/java/org/openmrs/util/databasechange/MigrateAllergiesChangeSet.java [hunk 1]
```diff
@@ -119,8 +119,8 @@ else if (AllergySeverity.SEVERE.name().equals(severity)) {
 				if (allergyType == null) {
 					allergyType = "DRUG";
 				}
-				else if ("ENVIRONMENTAL".equals(allergyType)) {
-					allergyType = "ENVIRONMENT";
+				else if ("OTHER".equals(allergyType)) {
+					allergyType = "OTHER";
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/databasechange/MigrateAllergiesChangeSet.java]: What can you explain what exactly is being accomplished by this last else if?

Comment (reubenvarghese1) [inline api/src/main/java/org/openmrs/util/databasechange/MigrateAllergiesChangeSet.java]: @dkayiwa Nothing at all! Thanks for pointing out my foolishness yet again! ;-)

**Comment by dkayiwa** [comment]
Some existing implementations stored allergies that look like these: https://github.com/openmrs/openmrs-core/blob/1.9.x/api/src/main/java/org/openmrs/activelist/AllergyType.java
You need to take care of upgrades from such.

**Comment by djazayeri** [comment]
I'm confused by this changeset and PR. This RA ticket should not require any changes to openmrs-core.

The idea is that openmrs-core supports these AllergenTypes, but that the UI was designed for only 3. We want to leave the UI basically looking the same as it did with the release of RefApp 2.4, but it needs to handle a different back-end structure.

**Comment by djazayeri** [comment]
PS- I think this PR should be closed, insofar as it's about RA-1283, and not about a core ticket.

**Pull request closed without merging**